### DISCUSSION TITLE
The npm run dev preview of the  Vite project failed

### DISCUSSION THREAD

**Discussion by sanbei101** [coder/code-server#6888] [Q&A] [answered]
```
root@uth1qb8ng6d3eyzi:~/nextjs/test-1# npm run dev

> test-1@0.0.0 dev
> vite


  VITE v5.3.3  ready in 353 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
```
![image](https://github.com/coder/code-server/assets/95988545/9a9e34eb-7ea1-4d7b-90a9-b2a22e2d2691)
![image](https://github.com/coder/code-server/assets/95988545/0ca9e97b-4743-47fa-a444-034344cf8726)
```
请求 URL:
http://125.91.126.101:8000/@vite/client
请求方法:
GET
状态代码:
404 Not Found
远程地址:
127.0.0.1:7897
引用站点策略:
strict-origin-when-cross-origin
```
我是第一次在code-server中开发前端项目,为什么会预览失败呢?[我的相关安全组端口全是打开的]
`I'm developing a front-end project for the first time in code-server, why does the preview fail? [All my security group ports are open]`

**Comment by code-asher** [comment, marked as answer]
Your application needs to be configured to prepend the proxy path to its requests (`/proxy/5173/` in your case) or it needs to make the requests relatively.  Currently it is making requests against the root, which is not right.

You might also have to use `/absproxy/5173` instead, depending on how the application is configured.  I am not familiar enough with Vite to say, but you can read https://github.com/coder/code-server/discussions/4541#discussioncomment-1722648

If you can, use the subdomain proxy instead.  It avoids these types of issues.

**Reply by sanbei101** [reply]
> ```
> import { defineConfig } from 'vite'
> import vue from '@vitejs/plugin-vue'
> 
> export default defineConfig({
>   plugins: [vue()],
>   server: {
>     port: 3454,
>   },
>   base: "/absproxy/3454",
> })
> ```

是的,当我这样设置之后就可以在`http://<myip>:8000/proxy/3454/absproxy/3454`上预览我的dev页面了!感谢你
